src/utils/application.data.ts

```diff
@@ -68,9 +68,22 @@ export const getFromApplicationData = (req: Request, key: string, id: string): a
   return appData[key][index];
 };
 
+export const getFromApplicationDataIfPresent = (req: Request, key: string, id: string): any => {
```

Comment:
Do we still need these functions `getFromApplicationDataIfPresent` and `getIndexInApplicationDataIfPresent` if we map BOs and Trusts IDs in the SDK?

Reply:
Quick one to improve tests, instead of duplicating the `getFromApplicationData` and `getIndexInApplicationData`, you could change the fingerprint of those functions by adding a `throwable` arg. Something like 

```
export const getFromApplicationData = (req: Request, key: string, id: string, throwable: boolean = true): any => {
  const appData: ApplicationData = getApplicationData(req.session);

  const index = getIndexInApplicationData(req, appData, key, id, throwable);
  if (index === -1 && throwable) {
    throw createAndLogErrorRequest(req, `application.data getFromApplicationData - unable to find object in session data for key ${key} and ID ${id}`);
  }
  return appData[key][index];
};

const getIndexInApplicationData = (req: Request, appData: ApplicationData, key: string, id: string, throwable: boolean = true) => {
  if (id && appData && appData[key]) {
    return appData[key].findIndex(object => object[ID] === id);
  }
  if (throwable) {
    throw createAndLogErrorRequest(req, `application.data getIndexInApplicationData - unable to find object in session data for key ${key} and ID ${id}`);
  }
};
```
and call the `getFromApplicationData` with the following args on trust information controller
```
getFromApplicationData(req, BeneficialOwnerIndividualKey, beneficialOwnerIds[i], false);
```